### PR TITLE
i18n(es): Update `styling` & `typescript`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### Description (required)

Updated `styling` & `typescript` based on these commits:
- 3894f296424da716c9d53bcf7c59473004e5f1f7
- efd5e737d312b169112dbaa6fcca81f3bc5bdd34

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/es/guides/styling.mdx [hunk 1]
```diff
@@ -132,11 +132,13 @@ En Astro, los atributos HTML tales como `class` no se pasan automáticamente a l
 
 En cambio, debes aceptar una prop `class` en el componente hijo y aplicársela al elemento raíz. Al desestructurar las props debes renombrarlo, porque `class` es una [palabra clave (o reservada)](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Lexical_grammar#palabras_clave) en JavaScript.
 
+Usando la estrategia por defecto de estilo acotado, también debes pasar el atributo `data-astro-cid-*`. Puedes hacer esto pasando el `...rest` de las props al componente. Si has cambiado `scopedStyleStrategy` a `'class'` o `'where'`, la prop `...rest` no es necesaria.
```
Comment (Waxer59) [inline src/content/docs/es/guides/styling.mdx:135]:
```suggestion
Usando la estrategia por defecto de estilo con ámbito, también debes pasar el atributo `data-astro-cid-*`. Puedes hacer esto pasando el `...rest` de las props al componente. Si has cambiado `scopedStyleStrategy` a `'class'` o `'where'`, la prop `...rest` no es necesaria.
```

File: src/content/docs/es/guides/styling.mdx [hunk 2]
```diff
@@ -153,10 +155,8 @@ import MyComponent from "../components/MyComponent.astro"
 <MyComponent class="red">¡Esto será rojo!</MyComponent>
 ```
 
-Este patrón te permite estilar componentes hijos directamente. Astro pasará el nombre de la clase local del padre (p. ej. `astro-hhnqfkh6`) por la propiedad `class` automaticamente, incluyendo el hijo en el ámbito de su padre. Ten en cuenta que este patrón solo funciona cuando tu [opción `scopedStyleStrategy`](/es/reference/configuration-reference/#scopedstylestrategy) está configurada como `'where'` o `'class'`.
-
 :::note[Clases globales de componentes padre]
```
Comment (Waxer59) [inline src/content/docs/es/guides/styling.mdx:158]:
```suggestion
:::note[Estilos con ámbito desde componentes padres]
```

File: src/content/docs/es/guides/styling.mdx [hunk 3]
```diff
@@ -153,10 +155,8 @@ import MyComponent from "../components/MyComponent.astro"
...
-Debido a que la propiedad `class` incluye al hijo en el ámbito de su padre, es posible que los estilos caigan en cascada de padre a hijo. Para evitar que esto tenga efectos secundarios no deseados, asegúrese de usar nombres de clase únicos en el componente secundario.
+Debido a que el atributo `data-astro-cid-*` incluye al hijo en el alcance de su padre, es posible que los estilos se filtren del padre al hijo. Para evitar que esto tenga efectos secundarios no deseados, asegúrate de usar nombres de clase únicos en el componente hijo.
```
Comment (Waxer59) [inline src/content/docs/es/guides/styling.mdx:159]:
```suggestion
Debido a que el atributo `data-astro-cid-*` incluye al hijo en el alcance de su padre, es posible que los estilos se propaguen del padre al hijo. Para evitar que esto tenga efectos secundarios no deseados, asegúrate de usar nombres de clase únicos en el componente hijo.
```

**Review by Waxer59 — COMMENTED** [review]
Buena PR 🤠, te dejo unas sugerencias!

**Comment by dreyfus92** [comment]
Muchas gracias por la revisión @Waxer59 😁

**Review by Waxer59 — APPROVED** [review]
LGTM ✅

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Good work 🙌

**Pull request merged**